src/wiringX.c

```diff
@@ -425,6 +428,181 @@ int wiringXSPISetup(int channel, int speed) {
 	return -1;
 }
 
+int wiringXSerialOpen(char *device, struct wiringXSerial_t wiringXSerial) {
+	struct termios options;
+	speed_t myBaud;
+	int status, fd;
+
+	switch(wiringXSerial.baud) {
+		case     50:	myBaud = B50; break;
+		case     75:	myBaud = B75; break;
+		case    110:	myBaud = B110; break;
+		case    134:	myBaud = B134; break;
+		case    150:	myBaud = B150; break;
+		case    200:	myBaud = B200; break;
+		case    300:	myBaud = B300; break;
+		case    600:	myBaud = B600; break;
+		case   1200:	myBaud = B1200; break;
+		case   1800:	myBaud = B1800; break;
+		case   2400:	myBaud = B2400; break;
+		case   4800:	myBaud = B4800; break;
+		case   9600:	myBaud = B9600; break;
+		case  19200:	myBaud = B19200; break;
+		case  38400:	myBaud = B38400; break;
+		case  57600:	myBaud = B57600; break;
+		case 115200:	myBaud = B115200; break;
+		case 230400:	myBaud = B230400; break;
+
+		default:
+		return -2;
+	}
+
+	if((fd = open(device, O_RDWR | O_NOCTTY | O_NDELAY | O_NONBLOCK)) == -1) {
+		return -1;
+	}
+
+	fcntl(fd, F_SETFL, O_RDWR);
+
+	tcgetattr(fd, &options);
+
+	cfmakeraw(&options);
+	cfsetispeed(&options, myBaud);
+	cfsetospeed(&options, myBaud);
+
+	options.c_cflag |= (CLOCAL | CREAD);
+
+	options.c_cflag &= ~CSIZE;
+	switch (wiringXSerial.databits) {
+		case 7:
+			options.c_cflag |= CS7;
+			break;
+		case 8:
+			options.c_cflag |= CS8;
+			break;
+		default:
+			wiringXLog(LOG_ERR, "Unsupported data size");
+			return (FALSE);
+	}
+	switch (wiringXSerial.parity) {
+		case 'n':
+		case 'N':/*no parity*/
+			options.c_cflag &= ~PARENB;
+			options.c_iflag &= ~INPCK;
+			break;
+		case 'o':
+		case 'O':
+			options.c_cflag |= (PARODD | PARENB);
+			options.c_iflag |= INPCK; /* Disable parity checking */
+			break;
+		case 'e':
+		case 'E':
+			options.c_cflag |= PARENB; /* Enable parity */
+			options.c_cflag &= ~PARODD;
+			options.c_iflag |= INPCK;
+			break;
+		case 'S':
+		case 's': /*as no parity*/
+			options.c_cflag &= ~PARENB;
+			options.c_cflag &= ~CSTOPB;
+			break;
+		default:
+			wiringXLog(LOG_ERR, "Unsupported parity");
+		return (FALSE);
+	}
+	switch (wiringXSerial.stopbits) {
+		case 1:
+			options.c_cflag &= ~CSTOPB;
+			break;
+		case 2:
+			options.c_cflag |= CSTOPB;
+			break;
+		default:
+			wiringXLog(LOG_ERR, "Unsupported data stop bits");
+		return (FALSE);
+	}
+	switch (wiringXSerial.flowcontrol) {
+		case 'x':
+		case 'X':
+			options.c_iflag |= (IXON | IXOFF | IXANY);
+			break;
+		case 'n':
+		case 'N':
+			options.c_iflag &= ~(IXON | IXOFF | IXANY);
+			break;
+		default:
+			wiringXLog(LOG_ERR, "Unsupported data flow control");
+			return (FALSE);
+	}
+
+	options.c_lflag &= ~(ICANON | ECHO | ECHOE | ISIG);
+	options.c_oflag &= ~OPOST;
+
+	options.c_cc[VMIN] = 0;
+	options.c_cc[VTIME] = 150;
+
+	tcflush(fd,TCIFLUSH);
+
+	tcsetattr(fd, TCSANOW | TCSAFLUSH, &options);
+
+	ioctl(fd, TIOCMGET, &status);
+
+	status |= TIOCM_DTR;
+	status |= TIOCM_RTS;
+
+	ioctl(fd, TIOCMSET, &status);
+
+	usleep(10000);
```

Comment:
I see Gordon add a usleep in RaspberryPi serial. 
I tested on radxa, it's of no use and I'll remove it.